pkg/cmd/roachprod/install/cluster_synced.go

```diff
@@ -491,14 +521,26 @@ func (c *SyncedCluster) Run(stdout, stderr io.Writer, nodes []int, title, cmd st
 	if !stream {
 		for i, r := range results {
 			fmt.Fprintf(stdout, "  %2d: %s\n", nodes[i], r)
+			if errors[i] != nil {
+				if cmdKind == CockroachCmd {
+					errors[i] = rpErrors.ClassifyCockroachError(errors[i])
+				} else {
+					errors[i] = rpErrors.ClassifyCmdError(errors[i])
+				}
+				if err, ok := errors[i].(rpErrors.BaseError); ok {
+					fmt.Fprintln(stdout, err.LogString())
+				}
+			}
 		}
 	}
 
+	// TODO(jamesl): Choose the best error to return. SSH? cockroach? CMD? roachprod?
```

Comment:
By providing a deterministic hierarchy of result error types, it could make debugging easier. This is particularly relevant to roachtest. Is this desirable? Is there a best priority order? My guess would be: cockroach, ssh, roachprod, other cmd.